sqlmesh/utils/file.py

```diff
@@ -0,0 +1,124 @@
+import json
+import random
+import time
+import typing as t
+from contextlib import contextmanager
+from uuid import uuid4
+
+import fsspec
+
+class FileTransactionHandler:
+    """A transaction handler which wraps a file path."""
+
+    def __init__(self, path: str) -> None:
+        """Creates a new FileTransactionHandler.
+        
+        Args:
+            path: The path to lock. It should follow the format of a table name.
+                Something like snapshots/version=<ver>/<snapshot_id>
+        """
+        self.path = path
+        # TODO: pass in a filesystem, naturally
+        self._fs: fsspec.AbstractFileSystem = fsspec.filesystem("file")
```

Comment:
is it trivial to add gzip?

Reply:
https://filesystem-spec.readthedocs.io/en/latest/_modules/fsspec/compression.html?highlight=gzip#

Reply:
Yep should be!